### PR TITLE
Pivot cleanup

### PR DESCRIPTION
**Pull request by ecraig12345**
#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

Was supposed to be looking at Pivot for the purpose of converting it to use derived state, but I got distracted by general messiness in the component. So this PR fixes it up a bit.

The changes to things other than the core Pivot files are to replace usage of the deprecated `IPivotItemProps.linkText` with `headerText` (and I added a deprecation warning for `linkText`).

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/7989)

### CODE REVIEW

File: apps/fabric-website-resources/src/components/pages/ColorsPage.tsx [hunk 1]
```diff
@@ -122,13 +122,13 @@ export class ColorsPage extends BaseComponent<{}, IColorsPageState> {
           </p>
         </div>
         {/* Hello! You've found hidden functionality for generating a theme from an image. This uses Microsoft's
-          * Cognitive Vision API, documented here:
```
Comment (ecraig12345) [inline apps/fabric-website-resources/src/components/pages/ColorsPage.tsx]: Seems nobody had run prettier on this file...hence the random spacing changes

File: packages/utilities/src/memoize.ts [hunk 2]
```diff
@@ -40,9 +40,9 @@ export function resetMemoizations(): void {
 }
 
 /**
- * Memoize decorator to be used on class methods. Note that the "this" reference
- * will be inaccessible within a memoized method, given that a cached method's this
- * would not be instance specific.
+ * Memoize decorator to be used on class methods. WARNING: the `this` reference
```
Comment (cliffkoh) [inline packages/utilities/src/memoize.ts]: :+1:

File: packages/office-ui-fabric-react/src/components/Pivot/PivotItem.tsx [hunk 3]
```diff
@@ -3,6 +3,14 @@ import { BaseComponent, getNativeProps, divProperties } from '../../Utilities';
 import { IPivotItemProps } from './PivotItem.types';
 
 export class PivotItem extends BaseComponent<IPivotItemProps, {}> {
+  constructor(props: IPivotItemProps) {
+    super(props);
+
+    this._warnDeprecations({
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Pivot/PivotItem.tsx]: :+1:

File: packages/office-ui-fabric-react/src/components/Pivot/PivotItem.tsx [hunk 4]
```diff
@@ -3,6 +3,14 @@ import { BaseComponent, getNativeProps, divProperties } from '../../Utilities';
...
+      linkText: 'headerText'
+    });
```
Comment (cliffkoh) [inline packages/office-ui-fabric-react/src/components/Pivot/PivotItem.tsx]: Missing `initialSelectedKey`?

Comment (ecraig12345) [reply packages/office-ui-fabric-react/src/components/Pivot/PivotItem.tsx]: No, that's a prop on Pivot itself, not PivotItem

Comment (cliffkoh) [reply packages/office-ui-fabric-react/src/components/Pivot/PivotItem.tsx]: :+1:

**Pull request merged**